src/napari_omero/plugins/loaders.py

```diff
@@ -65,52 +65,57 @@ def omero_proxy_reader(
 
 
 def load_image_wrapper(image: ImageWrapper) -> List[LayerData]:
-    return [
-        load_omero_channel(image, channel, c)
-        for c, channel in enumerate(image.getChannels())
+    data = get_data_lazy(image)
+    meta = get_omero_metadata(image)
+    # contrast limits range ... not accessible from plugin interface
+    # win_min = channel.getWindowMin()
+    # win_max = channel.getWindowMax()
+    return [(data, meta)]
+
+
+def get_omero_metadata(image: ImageWrapper) -> Dict:
+    """Get metadata from OMERO as a Dict to pass to napari."""
+    channels = image.getChannels()
+
+    colors = []
+    for ch in channels:
+        # use current rendering settings from OMERO
+        color = ch.getColor().getRGB()
+        color = [r / 256 for r in color]
+        colors.append(Colormap([[0, 0, 0], color]))
+
+    contrast_limits = [
+        [ch.getWindowStart(), ch.getWindowEnd()] for ch in channels
     ]
 
+    visibles = [ch.isActive() for ch in channels]
+    names = [ch.getLabel() for ch in channels]
 
-def load_omero_channel(
-    image: ImageWrapper, channel: ChannelWrapper, c_index: int
-) -> LayerData:
-    data = get_data_lazy(image, c_index=c_index)
-    color = channel.getColor().getRGB()
-    color = [r / 256 for r in color]
-    cmap = Colormap([[0, 0, 0], color])
     scale = None
-
-    # FIXME: still getting size mismatch sometimes  is there a getNDim()?
     if image.getSizeZ() > 1:
         size_x = image.getPixelSizeX()
         size_z = image.getPixelSizeZ()
         if size_x is not None and size_z is not None:
-            if image.getSizeT() > 1:
-                scale = [1, size_z / size_x, 1, 1]
-            else:
-                scale = [size_z / size_x, 1, 1]
-
-    meta = {
-        "blending": "additive",
-        "colormap": ("from_omero", cmap),
-        "scale": scale,
-        "name": channel.getLabel(),
-        "visible": channel.isActive(),
-        "contrast_limits": [channel.getWindowStart(), channel.getWindowEnd()],
+            scale = [1, size_z / size_x, 1, 1]
+
+    return {
+        'channel_axis': 1,
+        'colormap': colors,
+        'contrast_limits': contrast_limits,
+        'name': names,
+        'visible': visibles,
+        'scale': scale,
     }
-    # contrast limits range ... not accessible from plugin interface
-    # win_min = channel.getWindowMin()
-    # win_max = channel.getWindowMax()
-    return (data, meta)
 
 
 @timer
-def get_data_lazy(image: ImageWrapper, c_index: int = 0) -> da.Array:
-    """Get n-dimensional dask array, with delayed reading from OMERO image."""
+def get_data_lazy(image: ImageWrapper) -> da.Array:
```

Comment:
proposal for trimming this down a bit?
```python
@timer
def get_data_lazy(image: ImageWrapper) -> da.Array:
    """Get 5D dask array, with delayed reading from OMERO image."""
    nt, nc, nz, ny, nx = [getattr(image, f'getSize{x}')() for x in 'TCZYX']
    pixels = image.getPrimaryPixels()
    dtype = PIXEL_TYPES.get(pixels.getPixelsType().value, None)
    get_plane = delayed(timer(lambda idx: pixels.getPlane(*idx)))

    def get_lazy_plane(zct):
        return da.from_delayed(get_plane(zct), shape=(ny, nx), dtype=dtype)

    # 5D stack: TCZXY
    t_stacks = []
    for t in range(nt):
        c_stacks = []
        for c in range(nc):
            z_stack = []
            for z in range(nz):
                z_stack.append(get_lazy_plane((z, c, t)))
            c_stacks.append(da.stack(z_stack))
        t_stacks.append(da.stack(c_stacks))
    return da.stack(t_stacks)
```

Reply:
will continue this discussion on https://github.com/ome/omero-py/issues/227